### ISSUE TITLE
Syncing GCP repositories

### ISSUE THREAD

**Issue by tomplus** [xelalexv/dregsy#10]
I'm trying to sync docker images from the Google Container Registry (us.gcr.io) using a service account (https://cloud.google.com/container-registry/docs/advanced-authentication). There is "auth" with encoded json:

```
_json_key:{
  "type": "service_account",
  "project_id": "...
  ...
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://oauth2.googleapis.com/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
 ...
```

Any chance to set up synchronization with GCP repos? I tried skopeo and docker driver without success.

My errors:

```
2019-09-06T10:22:38Z [ERROR] invalid character '_' looking for beginning of value
2019-09-06T10:22:39Z [ERROR] error listing image tags: time="2019-09-06T10:22:39Z" level=fatal msg="Error reading manifest latest in us.gcr.io/xxx: unauthorized: You don't have the needed permissions to perform this operation, and you may have invalid credentials. To authenticate your request, follow the steps in: https://cloud.google.com/container-registry/docs/advanced-authentication" 
```

and with docker as relay:

```
2019-09-06T10:23:35Z [ERROR] error pulling source image 'us.gcr.io/xxx': Error response from daemon: pull access denied for us.gcr.io/xxx, repository does not exist or may require 'docker login'

docker daemon:

denied: Failed to read tags for host 'us.gcr.io', repository '/v2/xxx/tags/list'"
```

Thanks.

**Comment by xelalexv** [comment]
Thanks for the report & sorry for the late reply! At the moment, GCR with authentication is not supported. I'm sure it can be added, but I don't have the bandwidth to look into this right now.

How are you running dregsy - native, dockerized, or k8s? On the page you referenced, the last paragraph says:

> Compute Engine instances and Google Kubernetes Engine clusters can push and pull Container Registry images based on Cloud Storage scopes on the instances. 

If I interpret this correctly, then running dregsy on a Compute Engine instance or GKE with appropriate Cloud Storage scopes set would solve the authentication problem, but I don't have much experience with GCP.

**Comment by tomplus** [comment]
Thanks for your reply. I tried to run it on-prem but you are right it sounds like a good workaround.

**Comment by xelalexv** [comment]
implemented with PR #30
